pkgs/development/libraries/cairo/default.nix

```diff
@@ -1,171 +1,121 @@
-      hash = "sha256-mbTg67e7APfdELsuMAgXdY3xokWbGtHF7VDD5UyYqKM=";
-    })
-
-  ];
-
   outputs = [ "out" "dev" "devdoc" ];
   outputBin = "dev"; # very small
+
   separateDebugInfo = true;
 
   nativeBuildInputs = [
+    docbook-xsl-nons
+    gtk-doc
+    meson
+    ninja
     pkg-config
+    python3
   ];
 
-  buildInputs = [
-    libiconv
-    libintl
-  ] ++ optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
+  buildInputs = lib.optionals stdenv.isDarwin (with darwin.apple_sdk.frameworks; [
     CoreGraphics
     CoreText
     ApplicationServices
     Carbon
   ]);
 
-  propagatedBuildInputs = [ fontconfig expat freetype pixman zlib libpng ]
-    ++ optionals x11Support [ libXext libXrender ]
-    ++ optionals xcbSupport [ libxcb xcbutil ]
-    ++ optional gobjectSupport glib
-    ++ optional glSupport libGL
-    ; # TODO: maybe liblzo but what would it be for here?
-
-  configureFlags = [
-    "--enable-tee"
-  ] ++ (if stdenv.isDarwin then [
-    "--disable-dependency-tracking"
-    "--enable-quartz"
-    "--enable-quartz-font"
-    "--enable-quartz-image"
-    "--enable-ft"
-  ] else (optional xcbSupport "--enable-xcb"
-    ++ optional glSupport "--enable-gl"
-    ++ optional pdfSupport "--enable-pdf"
-  )) ++ optional (!x11Support) "--disable-xlib";
-
-  preConfigure =
-  # On FreeBSD, `-ldl' doesn't exist.
-    lib.optionalString stdenv.isFreeBSD
-       '' for i in "util/"*"/Makefile.in" boilerplate/Makefile.in
-          do
-            cat "$i" | sed -es/-ldl//g > t
-            mv t "$i"
-          done
-       ''
-    +
-    ''
-    # Work around broken `Requires.private' that prevents Freetype
-    # `-I' flags to be propagated.
-    sed -i "src/cairo.pc.in" \
-        -es'|^Cflags:\(.*\)$|Cflags: \1 -I${freetype.dev}/include/freetype2 -I${freetype.dev}/include|g'
-    substituteInPlace configure --replace strings $STRINGS
-    '';
-
-  enableParallelBuilding = true;
-
-  doCheck = false; # fails
+  propagatedBuildInputs = [
+    fontconfig
+    freetype
+    glib
+    libpng
+    lzo
+    pixman
+    zlib
+  ] ++ lib.optionals x11Support [
+    libX11
+    libXext
+    libXrender
+  ] ++ lib.optionals xcbSupport [
+    libxcb
+  ];
 
-  postInstall = lib.optionalString stdenv.isDarwin glib.flattenInclude;
+  mesonFlags = [
+    "-Dgtk_doc=true"
+    "-Dspectre=disabled" # only useful for tests
+    "-Dsymbol-lookup=disabled"
+    "-Dtests=disabled"
+    (lib.mesonEnable "xlib" x11Support)
+    (lib.mesonEnable "xcb" xcbSupport)
+  ] ++ (
+    # The meson-cc-tests/ipc_rmid_deferred_release.c test program
+    # won't do its job when cross compiling.
+    let
+      crossFile = writeText "cross-file.conf" ''
+        [properties]
+        ipc_rmid_deferred_release = 'false'
+      '';
+    in
+    lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform) [
+      "--cross-file=${crossFile}"
+    ]
+  );
+
+  postPatch = ''
+    patchShebangs version.py
+  '';
```

Comment:
This script simply returns the apparent version number, and we can take control of this with:

```
  postPatch = let
    # despite the suffix this is not a python script - this allows
    # us to take control of the apparent version and avoid the
    # direct python dependency at the same time
    dummyVersionScript = writeShellScript "version.py" ''
      echo '${finalAttrs.version}'
    '';
  in ''
    rm version.py
    ln -s ${dummyVersionScript} version.py
  '';
```

don't know if it's really worth the fiddle though